File: 209-minimum-size-subarray-sum/209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, t: int, v: List[int]) -> int:
            if sum(v) < t :
                return 0
            l , r = 0 , 1
            sm = v[0]
            ans = len(v)
            if sm >= t:
                return 1
            while l <= r and r < len(v) and ans !=1:
                sm += v[r]
                while sm >= t:
                    ans = min(ans , r - l + 1)
                    if ans == 1 :break
                    sm -= v[l]
                    l +=1
                r +=1

            return ans

File: 209-minimum-size-subarray-sum/209-minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        
	int n = nums.size();
	if(n == 1){
		if(nums[0] >= target)
			return 1;
		return 0;
	}

	int sm = nums[0];
	if(sm >= target)
		return 1;

	int l = 0 , r = 1;
	int ans = n ;
	bool ok = false;
	while(l <= r && r < n){

		sm += nums[r];
		while(l<= r && r < n && sm >= target){
			ok = true;
			sm -= nums[l];
			ans = min(ans , r - l + 1);
			l++;
		}

		r++;
	}

	if(!ok)return 0;
	return ans;
    }
};